framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/AsyncResult.cs

```diff
@@ -0,0 +1,238 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//----------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Samples.AzureQueueStorage
+{
+    /// <summary>
+    /// A generic base class for IAsyncResult implementations
+    /// that wraps a ManualResetEvent.
+    /// </summary>
+    abstract class AsyncResult : IAsyncResult
+    {
```

Comment:
I think you can get rid of this file. After you fix the BeginSend and EndSend method to defer to an async Task based method, the only usage here is the CompletedAsyncResult class. With that you can replace it with `return Task.CompletedTask.ToApm(callback, state);` and `result.ToApmEnd();` in the end method. Then this file should go away.

Reply:
That make the code a lot simpler.